tasks/sast-go.yaml

```diff
@@ -0,0 +1,39 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: gosec-check
+spec:
+  workspaces:
+    - name: workspace
+  results:
+    - description: Test output
+      name: HACBS_TEST_OUTPUT
+  steps:
+    - name: gosec-check
+      image: quay.io/redhat-appstudio/hacbs-test:stable
+      workingDir: $(workspaces.workspace.path)/hacbs/$(context.task.name)
+      script: |
+        /usr/local/go/bin/gosec -no-fail -fmt=sarif -out=gosec_output.json $(workspaces.workspace.path)/... | tee gosec_output.txt
+        GOSEC_EXIT_CODE=$?
+
+        # Test if any package was found
+        # Even with -no-fail, gosec uses exit code 1 for several states,
+        # including when there are no packages found.
+        grep "No packages found$" gosec_output.txt && echo "Exit 0 because no packages found" && exit 0
+        exit $GOSEC_EXIT_CODE
+
+    - name: test-format-result
```

Comment:
I do not see reason to have this as separate step. BTW proposing to add this script to quay.io/redhat-appstudio/hacbs-test image so then it will be easier to reuse.